src/js/ports.js

```diff
@@ -228,204 +232,184 @@ function subscribeToCTokenPorts(app, eth) {
-  app.ports.askCTokenGetBalancesPort.subscribe(
-    ({ blockNumber, customerAddress, cTokens: cTokenEntries, compoundLens }) => {
-      const CompoundLens = getContractJsonByName(eth, 'CompoundLens');
+  // port queryAllWithAccountPort : { blockNumber : Int, customerAddress : String, cTokens : List ( String, CTokenPortData ), compAddress: String, capFactoryAddress: String } -> Cmd msg
+  app.ports.queryAllWithAccountPort.subscribe(
+    async({ blockNumber, customerAddress, cTokens: cTokenEntries, compAddress, capFactoryAddress }) => {
       let cTokens = supportFromEntries(cTokenEntries);
 
-      wrapCall(
-        app,
-        eth,
-        [[CompoundLens, compoundLens, 'cTokenBalancesAll', [Object.keys(cTokens), customerAddress]]],
-        blockNumber
-      )
-        .then(([results]) => {
-          const cTokenBalancesList = results.map(
-            ({
-              cToken: cTokenAddress,
-              balanceOf: cTokenWalletBalanceResult,
-              borrowBalanceCurrent: underlyingBorrowBalanceResult,
-              balanceOfUnderlying: underlyingSupplyBalanceResult,
-              tokenBalance: tokenBalanceResult,
-              tokenAllowance: tokenAllowanceResult,
-            }) => {
-              let { underlyingAssetAddress, underlyingDecimals, cTokenDecimals, cTokenSymbol } = cTokens[
-                cTokenAddress.toLowerCase()
-              ];
-              const walletBalance = toScaledDecimal(parseWeiStr(cTokenWalletBalanceResult), cTokenDecimals);
-              const borrowBalance = toScaledDecimal(parseWeiStr(underlyingBorrowBalanceResult), underlyingDecimals);
-              const supplyBalance = toScaledDecimal(parseWeiStr(underlyingSupplyBalanceResult), underlyingDecimals);
-              const tokenBalance = toScaledDecimal(tokenBalanceResult, underlyingDecimals);
-              const tokenAllowance = toScaledDecimal(tokenAllowanceResult, underlyingDecimals);
-
-              if (cTokenSymbol == 'cETH') {
-                // Since we're on eth anyway
-                app.ports.giveAccountBalancePort.send({
-                  balance: tokenBalance,
-                });
-              }
+      const web3 = await withWeb3Eth(eth);
+
+      const QUERY = Sleuth.querySol(SleuthQuery, { queryFunctionName: 'queryAllWithAccount' });
 
-              return {
-                cTokenAddress: cTokenAddress,
-                customerAddress: customerAddress,
-                cTokenWalletBalance: walletBalance,
-                underlyingAssetAddress: underlyingAssetAddress,
-                underlyingBorrowBalance: borrowBalance,
-                underlyingSupplyBalance: supplyBalance,
-                underlyingTokenWalletBalance: tokenBalance,
-                underlyingTokenAllowance: tokenAllowance,
-              };
+      const provider = new StaticJsonRpcProvider(web3.currentProvider.host);
+      let sleuth = new Sleuth(provider);
+
+      Promise.all([
+        getTransactionCount(eth, customerAddress), 
```

Comment:
These are the fun little quirks for things we need to consider if we ever want to move off of web3-reading. Good callout that this is here.